[PDI-14950] Spoon does not pass instanceName parameter to MSSQL Server JDBC connection string

IvanNikolaychuk opened the pull request:
The reason why this bug occures is [this line] (https://github.com/pentaho/pentaho-kettle/blob/master/core/src/org/pentaho/di/core/database/DatabaseMeta.java#L2717).
We set extra option, specifying MSSQL (that is a hardcoded string) as databaseTypeCode in all cases, though it can be of type MSSQLNATIVE also. 
My first approach was to treat this plugins as seperate databaseTypeCode, but it's not such an easy task as it seemed to be. In short: I could not do it, because when edditing/creating a database we can't pull an up-to-date databasePluginId *before* clicking on "OK" button. I can provide a more detailed description for the problems of this approach, if necessary.

My second approach is described bellow in a short form and more explisit in java docs.

When generating connection URL, treat the one connection type B as connection type A if:
 1. A is B
 2. B is subset of A, where every connection setting, that is valid to B is valid to A also.

 See javaDocks for databaseForBothConnTypesIsTheSame method for more details.


@akhayrutdinov, could you please review it?

akhayrutdinov commented:
@brosander, can you please take a look? I didn't find time for it today. Thanks

IvanNikolaychuk commented:
@mtseu, could you please review it?

IvanNikolaychuk commented:
@mtseu, @akhayrutdinov, @brosander, could you please review this PR?

lucboudreau commented:
http://10.100.11.222:8080/job/wingman-osgi/52/

lucboudreau commented:
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/database/DatabaseMeta.java
  core/test-src/org/pentaho/di/core/database/DatabaseMetaTest.java

Newly Broken Tests:
======================
org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: expected:<0> but was:<-1>
	at org.pentaho.di.core.row.value.ValueMetaStringTest.assertSignum(ValueMetaStringTest.java:831)
	at org.pentaho.di.core.row.value.ValueMetaStringTest.assertSignum(ValueMetaStringTest.java:818)
	at org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull(ValueMetaStringTest.java:518)

```
org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: expected:<true> but was:<false>
	at org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull(ValueMetaStringTest.java:180)

```

Unit test coverage change:
============
org.pentaho.di.core.database.DatabaseMeta:
 * Branch Change :  + 1.20%
 * Complexity Change :  + .60%
 * Instruction Change :  + .28%
 * Line Change :  + .35%
 * Method Change :  + .34%

org.pentaho.di.core.row.value.ValueMetaBase:
 * Branch Change : **-.08%**
 * Complexity Change : **-.10%**
 * Instruction Change : **-.02%**
 * Line Change : **-.05%**



Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards
File /home/buildguy/project/core/src/org/pentaho/di/core/database/DatabaseMeta.java

1. Message: ')' is not preceded with whitespace. at Line: 1,184 Column: 92

File /home/buildguy/project/core/test-src/org/pentaho/di/core/database/DatabaseMetaTest.java

1. Message: '=' is not preceded with whitespace. at Line: 55 Column: 62

brosander commented:
@IvanNikolaychuk can you retag it as [CLEANUP]? we have reporting expecting known tags

brosander commented:
@IvanNikolaychuk also, it looks like the original commit is in there now, so probably just using your issue number would work

IvanNikolaychuk commented:
@brosander, I squashed it into one commit, as there is no need for seperate CLEANUP commit - all the checkstyle violations above were brought with the new code that I added in this PR.

lucboudreau commented:
http://10.100.11.222:8080/job/wingman/1/

lucboudreau commented:
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/database/DatabaseMeta.java
  core/test-src/org/pentaho/di/core/database/DatabaseMetaTest.java

Still Broken Tests:
======================
org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: expected:<0> but was:<-1>
	at org.pentaho.di.core.row.value.ValueMetaStringTest.assertSignum(ValueMetaStringTest.java:831)
	at org.pentaho.di.core.row.value.ValueMetaStringTest.assertSignum(ValueMetaStringTest.java:818)
	at org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull(ValueMetaStringTest.java:518)

```
org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: expected:<true> but was:<false>
	at org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull(ValueMetaStringTest.java:180)

```

Unit test coverage change:
============
org.pentaho.di.core.database.DatabaseMeta:
 * Branch Change :  + 1.20%
 * Complexity Change :  + .60%
 * Instruction Change :  + .28%
 * Line Change :  + .35%
 * Method Change :  + .34%
